### PR TITLE
Playground frontend in React

### PR DESCRIPTION
**Pull request by torhovland**
Implements a basic version of #171.

A version without sample input and query is already deployed to https://topiary.tweag.io/playground.

- Move wasm-app to subdir.
- Working playground in React.
- favicon, etc.
- Layout.
- Button style.
- Switch to TypeScript.
- Move Tree-sitter files.
- Load Tree-sitter.
- Rename script.
- Remove prototype web playground.
- Build React app in CI.
- Sample input and query.

### CODE REVIEW

File: .github/workflows/ci.yml
Comment (Xophmeister) [inline .github/workflows/ci.yml:1]: Not for this PR, but does the building of the WASM app need to happen on both Ubuntu and macOS? On the one hand, it proves it works on macOS, but isn't the main utility here to stage for deployment, which only needs to be done once?

File: web-playground/react-app/README.md
Comment (Xophmeister) [inline web-playground/react-app/README.md:1]: I guess it doesn't hurt to keep this vendored `README`... (You see it all the time. It wouldn't surprise me if Facebook's Ts&Cs mandated it :sweat_smile:)

**Review by Xophmeister — APPROVED** [review]

File: web-playground/react-app/README.md
Comment (torhovland) [reply web-playground/react-app/README.md:1]: Yeah, I tried to modify the template as little as possible. We can always delete more later.

File: .github/workflows/ci.yml
Comment (torhovland) [reply .github/workflows/ci.yml:1]: You're right. It's important to check that the CLI app builds on multiple architectures, but it doesn't really matter for the playground.

**Comment by Xophmeister** [comment]
Ah nuts: I resolved the conversations and it auto-merged... Let me restore the branch, at least.

**Pull request merged**